community/kernel/src/main/java/org/neo4j/kernel/api/schema/index/IndexDescriptor.java

```diff
@@ -146,18 +196,14 @@ public String toString()
         return userDescription( SchemaUtil.idTokenNameLookup );
     }
 
-    /**
-     * Sorts indexes by type, returning first GENERAL indexes, followed by UNIQUE. Implementation is not suitable in
-     * hot path.
-     *
-     * @param indexes Indexes to sort
-     * @return sorted indexes
-     */
-    public static Iterator<SchemaIndexDescriptor> sortByType( Iterator<SchemaIndexDescriptor> indexes )
+    public StoreIndexDescriptor withId( long id )
+    {
+        return new StoreIndexDescriptor( this, id );
+    }
+
+    public StoreIndexDescriptor withIds( long id, long owningConstraintId )
     {
-        List<SchemaIndexDescriptor> materialized = Iterators.asList( indexes );
-        return Iterators.concat(
-                Iterators.filter( GENERAL, materialized.iterator() ),
-                Iterators.filter( UNIQUE, materialized.iterator() ) );
+        assert owningConstraintId >= 0;
```

Comment:
I agree in this case. For us who weren't in that recent discussion, what is the summary?

Reply:
In most cases there is no extra cost in a real Exception and it is a much safer guard since it is actually included in production code.
If we in production slip through a check only because we used assert instead of exception we end up in a situation that is harder to debug. Fail fast.
I think that was the gist of it.